### ISSUE TITLE
Command line Mame output mode (especially useful for chess ROM emulation)?

### ISSUE THREAD

**Issue by sshivaji** [mamedev/mame#2523]
If you look at some of the chess computer roms, e.g. glasgow.cpp or mephisto.cpp, and run one of them, its usually typing in something like "e2e4" and getting back a reply like "e7e5" on a big red LED display. Then, you can type in the next move and so on.. 

How can we convert this to a simple command line application without any video, that takes stdin input and outputs the LED value to stdout? Specifically, we just need the output of output().set_digit_value(..) to be written back to the console. 

I am looking for the simplest way to do this via Lua and a tiny build. The chess branch of my tree - https://github.com/sshivaji/mame/tree/chess has a tiny build for just the chess modules. 

In the past, a few people have made an attempt to modify mame.c to get this functionality. I have stored one of the patch attempts on mame-0138 here - https://github.com/sshivaji/chessroms/tree/master/Source_WBMessMephisto_04_07_2010 

In summary, I would like a way to disable video mode, disable even audio, and just get command line output for chess roms. I also would like to do this without modifying mame.c or changing the drivers too much. Ideally something in Lua or a solution that is easy to maintain while the mame tree keeps updating to newer versions. 

A less ideal solution is perhaps to have glasgow.cpp and mephisto.cpp output the moves to a logger method. We can maybe run this with -screen None mode.

Would appreciate suggestions on this.

Thanks, Shiv

**Comment by sronco** [comment]
Hi,

this is not exactly what you ask, but a couple of months ago I had the strange idea of writing a Lua plugin to interface emulated chess machine with UCI/XBoard GUI, unfortunately I've never finished it and currently only supports a couple of drivers and it's only tested on Linux.

If you want to try it I created a repo [here](https://github.com/sronco/mame-chessengine) with what I have, maybe you can adapt it to meet your needs.

Anyway after installing the plugin you can play against the emulated Saitek RISC 2500 in XBoard
	xboard -fcp "./mame64 -skip_gameinfo -plugin chessengine risc"

or watch a match between two machines
	xboard -mm -fcp "./mame64 -skip_gameinfo -plugin chessengine ch2001" -scp "./mame64 -skip_gameinfo -plugin chessengine risc"

Hope this helps

Best regards,

Sandro

**Comment by sshivaji** [comment]
This is indeed extremely helpful!! I have forked your repo and will submit
patches.

My eventual goal is to make this work on the raspberry pi. Thanks a lot for
contributing your code. I think disabling video output is not necessary if
there is a lua solution.

For some reason, the lua plugin does not yet work. I am debugging the exact
cause. There is nothing in the logs.

Thanks!, Shiv

**Comment by rb6502** [comment]
I'd love to see this plugin finished up and included in MAME.

**Comment by sshivaji** [comment]
Thanks for suggesting this should be a mame official plugin. Would be great!

The lua script now seems to work. One needs lua 5.3 locally installed (I
assumed its embedded within mame). There are a bunch of minor things to
resolve, but the output seems decent enough to be a plugin even at this
point. I will experiment with a few more engines and post back.
